app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -149,4 +149,24 @@ def relation_values(association, target)
 
     top_level && (target == true || target.nil? || parent == target) ? :use_association : []
   end
+
+  def get_cluster(ems, cluster_hash, rp_hash, dc_hash)
+    cluster = EmsCluster.find_by(:ems_ref => cluster_hash[:ems_ref])
+    if cluster.nil?
+      rp = ems.resource_pools.build(rp_hash)
+      rp.save!
```

Comment:
If you are going to save it immediately, then instead of `build/save!` just call `create!`